pyanalyze/value.py

```diff
@@ -2184,21 +2183,22 @@ def unite_values(*values: Value) -> Value:
         else:
             subvals = [value]
         for subval in subvals:
-            if _is_unreachable(subval):
-                saw_unreachable = True
-                continue
             try:
                 # Don't readd it to preserve original ordering.
                 if subval not in hashable_vals:
                     hashable_vals[subval] = None
             except Exception:
                 unhashable_vals.append(subval)
     existing = list(hashable_vals) + unhashable_vals
-    num = len(existing)
+    reachabilities = [_is_unreachable(val) for val in existing]
```

Comment:
This defers the `_is_unreachable` call until *after* we have deduplicated the values, so we don't call it repeatedly if we have lots of duplicates.